lib/icingadb/icingadb-objects.cpp

```diff
@@ -1860,6 +1860,7 @@ void IcingaDB::SendStartedDowntime(const Downtime::Ptr& downtime)
 		"scheduled_end_time", Convert::ToString(TimestampToMilliseconds(downtime->GetEndTime())),
 		"has_been_cancelled", Convert::ToString((unsigned short)downtime->GetWasCancelled()),
 		"trigger_time", Convert::ToString(TimestampToMilliseconds(downtime->GetTriggerTime())),
+		"cancel_time", Convert::ToString(TimestampToMilliseconds(downtime->GetRemoveTime())),
```

Comment:
"would" is the most important word here. If for some reason this goes wrong, I want

* to know it via history
* no Icinga DB crash

Reply:
> no Icinga DB crash

That's the primary objective of this PR.

> * to know it via history

Know what? I mean a possible alternative could be to just remove `has_been_cancelled` from the start event altogether as that doesn't really make sense here (note: I didn't check/test how Icinga DB will behave in that case) and should be set in the end event. The downtime start and end events affect the same row in `downtime_history` anyways, so you can't tell which of both even set `cancel_time`.

Reply:
Nope. `2023-12-19 15:09:14 2023-12-19T14:09:14.260Z    FATAL   icingadb        Error 1048 (23000): Column 'has_been_cancelled' cannot be null`

Reply:
That's because the DB column is not nullable. How about just sending a hardcoded `0` for `has_been_cancelled` then? I mean, this is a downtime start event, we don't need to set this to a meaningful value as it is guaranteed to be synchronised with the downtime end event.
https://github.com/Icinga/icinga2/blob/949d983a76c2f184e7e8482baa8c68625ff99e17/lib/icingadb/icingadb-objects.cpp#L1954-L1956

Reply:
This sounds so cheaty, that's literally fake info.